homeassistant/components/mqtt/humidifier.py

```diff
@@ -0,0 +1,488 @@
+        raise ValueError("target_humidity_max must be > target_humidity_min")
+    if config.get(CONF_TARGET_HUMIDITY_MAX) > 100:
+        raise ValueError("max_humidity must be <= 100")
+    conf_target_humidity_range_min = (
+        config.get(CONF_TARGET_HUMIDITY_RANGE_MIN)
+        if CONF_TARGET_HUMIDITY_RANGE_MIN in config
+        else config.get(CONF_TARGET_HUMIDITY_MIN)
+    )
+    conf_target_humidity_range_max = (
+        config.get(CONF_TARGET_HUMIDITY_RANGE_MAX)
+        if CONF_TARGET_HUMIDITY_RANGE_MAX in config
+        else config.get(CONF_TARGET_HUMIDITY_MAX)
+    )
+    if conf_target_humidity_range_min >= conf_target_humidity_range_max:
+        raise ValueError(
+            "target_humidity_range_max must be > target_humidity_range_min"
+        )
+
+    return config
+
+
+PLATFORM_SCHEMA = vol.All(
+    mqtt.MQTT_RW_PLATFORM_SCHEMA.extend(
+        {
+            vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+            vol.Optional(CONF_OPTIMISTIC, default=DEFAULT_OPTIMISTIC): cv.boolean,
+            vol.Optional(CONF_COMMAND_TEMPLATE): cv.template,
+            vol.Required(CONF_TARGET_HUMIDITY_COMMAND_TOPIC): mqtt.valid_publish_topic,
+            vol.Optional(CONF_TARGET_HUMIDITY_COMMAND_TEMPLATE): cv.template,
+            vol.Optional(CONF_TARGET_HUMIDITY_STATE_TOPIC): mqtt.valid_subscribe_topic,
+            vol.Optional(CONF_TARGET_HUMIDITY_STATE_TEMPLATE): cv.template,
+            # CONF_MODE_COMMAND_TOPIC and CONF_AVAIALABLE_MODES_LIST must be used together
+            vol.Inclusive(
+                CONF_MODE_COMMAND_TOPIC, "available_modes"
+            ): mqtt.valid_publish_topic,
+            vol.Inclusive(
+                CONF_AVAILABLE_MODES_LIST, "available_modes", default=[]
+            ): cv.ensure_list,
+            vol.Optional(CONF_MODE_COMMAND_TEMPLATE): cv.template,
+            vol.Optional(CONF_MODE_STATE_TOPIC): mqtt.valid_subscribe_topic,
+            vol.Optional(CONF_MODE_STATE_TEMPLATE): cv.template,
+            vol.Optional(
+                CONF_TARGET_HUMIDITY_MIN, default=DEFAULT_MIN_HUMIDITY
+            ): cv.positive_int,
+            vol.Optional(
+                CONF_TARGET_HUMIDITY_MAX, default=DEFAULT_MAX_HUMIDITY
+            ): cv.positive_int,
+            vol.Optional(CONF_TARGET_HUMIDITY_RANGE_MIN): cv.positive_int,
+            vol.Optional(CONF_TARGET_HUMIDITY_RANGE_MAX): cv.positive_int,
+            vol.Optional(
+                CONF_PAYLOAD_RESET_HUMIDITY, default=DEFAULT_PAYLOAD_RESET
+            ): cv.string,
+            vol.Optional(
+                CONF_PAYLOAD_RESET_MODE, default=DEFAULT_PAYLOAD_RESET
+            ): cv.string,
+            vol.Optional(CONF_PAYLOAD_OFF, default=DEFAULT_PAYLOAD_OFF): cv.string,
+            vol.Optional(CONF_PAYLOAD_ON, default=DEFAULT_PAYLOAD_ON): cv.string,
+            vol.Optional(CONF_STATE_VALUE_TEMPLATE): cv.template,
```

Comment:
Let's sort the keys in the schema 👍

Reply:
Done but left the vol.Inclusive's together